ksqldb-engine/src/main/java/io/confluent/ksql/util/SandboxedSharedKafkaStreamsRuntimeImpl.java

```diff
@@ -35,7 +42,17 @@ public SandboxedSharedKafkaStreamsRuntimeImpl(
         sharedRuntime.getKafkaStreamsBuilder(),
         getSandboxStreamsProperties(sharedRuntime)
     );
-
+    Collection<NamedTopology> namedTopologies = collocatedQueries.values()
+        .stream()
+        .map(BinPackedPersistentQueryMetadataImpl::getTopology)
+        .collect(Collectors.toSet());
+    if (!namedTopologies.containsAll(sharedRuntime.kafkaStreams.getAllTopologies())
+        || !sharedRuntime.kafkaStreams.getAllTopologies().containsAll(namedTopologies)) {
+      log.warn("Streams topologies and registered queries do not align");
```

Comment:
Maybe better printing the two collections in the entry as well?

Also is that ever going to be expected? If not should we log as error and force-crash the instance too?

Reply:
Good idea about printing the two collections. There is a chance it can happen when another query is in the process of being replaced. So we shouldn't stop validation for this alone.